### PR TITLE
refactor(cardinal): move store manager and log to package

### PR DESCRIPTION
**Pull request by jerargus**
Closes: #WORLD-321, #WORLD-326

## What is the purpose of the change

-Move the StoreManager and Logger structs out of the ecs package and into sub-packages. 
-Delete some entity management code that is no longer used anywhere.

With the StoreManager in a dedicated sub-package, we can now work on the ECB system by extending this targeted package, instead of adding more code to the ecs package.

## Testing and Verifying

Existing unit tests verify that there are no functional changes.

### CODE REVIEW

File: cardinal/ecs/storemanager/store_manager.go
Comment (technicallyty) [inline cardinal/ecs/storemanager/store_manager.go:1]: maybe `ecs/store/manager.go`?

File: cardinal/ecs/storemanager/store_manager.go [hunk 1]
```diff
@@ -9,22 +9,23 @@ import (
 	"pkg.world.dev/world-engine/cardinal/ecs/component"
 	"pkg.world.dev/world-engine/cardinal/ecs/entity"
 	"pkg.world.dev/world-engine/cardinal/ecs/filter"
+	"pkg.world.dev/world-engine/cardinal/ecs/log"
 	"pkg.world.dev/world-engine/cardinal/ecs/storage"
 )
 
 type StoreManager struct {
```
Comment (technicallyty) [inline cardinal/ecs/storemanager/store_manager.go:16]: maybe just `Manager` depending on naming conventions

**Review by pyrofolium — APPROVED** [review]
LGTM. I think I need this change to fix the reflection issue with cql routes. 

In order to get rid of reflection I need IComponentType to have a method: GetRawJsonData, and for this it needs to access the store manager.

File: cardinal/ecs/storemanager/store_manager.go
Comment (jerargus) [reply cardinal/ecs/storemanager/store_manager.go:1]: ok, renamed to store.Manager. I added a package comment to store and storage to help distinguish them.

File: cardinal/ecs/storemanager/store_manager.go (hunk 1 above)
Comment (jerargus) [reply cardinal/ecs/storemanager/store_manager.go]: done

**Pull request merged**